common/djangoapps/student/models.py

```diff
@@ -1709,6 +1709,8 @@ def upgrade_deadline(self):
             return None
 
         if self.dynamic_upgrade_deadline is not None:
+            if datetime.now(UTC) >= self.dynamic_upgrade_deadline:
+                return None
```

Comment:
Why move it here?

Reply:
I just added some docstrings and comments that hopefully answer this?